api/lambda/authorizer.js

```diff
@@ -315,70 +318,128 @@ AuthPolicy.prototype = (function AuthPolicyClass() {
   };
 }());
 
-exports.Custom = (event, context, callback) => {
-  //console.log('Client authorization token:', event.authorizationToken);
-  //console.log('Method ARN:', event.methodArn);
-
-  // TODO: Move this initial key lookup step into constructor for on-time run only.
-  // Download and store the JWT set for our Cognito User Pool
-  // let jwtKeySet = 'https://cognito-idp.' + config.AWS_REGION + '.amazonaws.com/' + env.config.USER_POOL_ID + '/.well-known/jwks.json';
-  // logger.info('jwtKeySetUrl:', jwtKeySet);
-
-
-  // Decode and validate the incoming token
-  var decoded = null;
-  try {
-    decoded = jwt.decode(event.authorizationToken);
-    logger.info('JWT Decoded', decoded);
-  } catch(e) {
-    callback(e);
+
+function processAuthRequest(event, tokenIssuer, callback) {
+  var token = event.authorizationToken;
+
+  //Fail if the token is not jwt
+  var decodedJwt = jwt.decode(token, {complete: true});
+  if (!decodedJwt) {
+    console.log('Not a valid JWT token');
+    callback('Unauthorized');
+    return;
   }
 
-  const principalId = decoded.sub;
-  let admin = null;
-  if (decoded['custom:admin'] && decoded['custom:admin'] === 'true') {
-    admin = true;
+  //Fail if token is not from your User Pool
+  if (decodedJwt.payload['iss'] != tokenIssuer) {
+    console.log("invalid Issuer");
+    callback("Unauthorized");
+    return;
   }
 
-  //TODO(Justin): Validate signature against Cognito User Pool
-  //TODO(Justin): Validate expiration date/time
+  //Reject the jwt if it's not an 'Access Token'
```

Comment:
I believe you mean "identity token" here rather than "access token." At least that's what we used to use for our custom authorizer and "token_use = id" would imply the same thing.